tests/input/test_input_mouse.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from textual.app import App, ComposeResult
+from textual.geometry import Offset
+from textual.widgets import Input
+
+
+class InputApp(App[None]):
+    TEST_TEXT = "That gum you like is going to come back in style"
+
+    def compose(self) -> ComposeResult:
+        yield Input(self.TEST_TEXT)
+
+
+async def test_mouse_clicks_within():
+    """Mouse clicks should result in the cursor going to the right place."""
+    async with InputApp().run_test() as pilot:
+        for click_at, should_land in (
+            (0, 0),
+            (1, 1),
+            (10, 10),
+            (len(InputApp.TEST_TEXT) - 1, len(InputApp.TEST_TEXT) - 1),
+            (len(InputApp.TEST_TEXT), len(InputApp.TEST_TEXT)),
+            (len(InputApp.TEST_TEXT) * 2, len(InputApp.TEST_TEXT)),
+        ):
+            # Note the offsets to take into account the decoration around an
+            # Input.
+            await pilot.click(Input, Offset(click_at + 3, 1))
+            await pilot.pause()
+            assert pilot.app.query_one(Input).cursor_position == should_land
```

Comment:
What's the difference between having the test like this or using `pytest.mark.parametrize` to parametrise the `click_at` and `should_land` coordinates?

Reply:
Most likely my ignorance of `pytest` and a love of writing clear code that doesn't use magic. ;-)

Reply:
Well, I asked because there might be a genuine advantage to writing it like this.
What I _know_ is good about parametrising the tests is that if only part of these assertions fail, it is easier to figure out which one(s) did fail.

Reply:
I just tried it out after you asked and you GET MORE TESTS FROM YOUR TESTS so I'm sold. ¯\\_(ツ)_/¯ 

(technically I guess one advantage of the ignorant-of-pytest approach would be be a bit of extra speed, as you're not spinning up and ripping down an application for every parameter).

Reply:
Yeah x2
You get more tests for the same number of tests! 😆 and you also spend some more time, true.